File: src/dataset/cpp/rotational-cipher_exercism-toemm/claude/ClaudeSonnet4_rotational_cipher_v1.cpp
#include "rotational_cipher.h"

namespace rotational_cipher {

std::string rotate(std::string in, int shift) {
    shift = ((shift % 26) + 26) % 26;
    
    for (char& ch : in) {
        if (ch >= 'A' && ch <= 'Z') {
            ch = 'A' + (ch - 'A' + shift) % 26;
        } else if (ch >= 'a' && ch <= 'z') {
            ch = 'a' + (ch - 'a' + shift) % 26;
        }
    }
    
    return in;
}

}  // namespace rotational_cipher

File: src/dataset/Java/rotational-cipher_exercism-johannesrave-education/claude/ClaudeSonnet4_RotationalCipher_v1.java
class RotationalCipher {

    private final int shiftKey;

    RotationalCipher(int shiftKey) {
        this.shiftKey = shiftKey % 26; 
    }

    String rotate(String data) { 
        if (data.isEmpty()) return data;
        
        StringBuilder result = new StringBuilder(data.length());
        
        for (int i = 0; i < data.length(); i++) {
            char c = data.charAt(i);
            result.append(shiftChar(c));
        }
        
        return result.toString();
    }

    private char shiftChar(char c) {
        if (c >= 'a' && c <= 'z') {
            return (char) ('a' + (c - 'a' + shiftKey) % 26);
        } else if (c >= 'A' && c <= 'Z') {
            return (char) ('A' + (c - 'A' + shiftKey) % 26);
        }
        return c;
    }
}